File: app/src/main/java/com/limetac/scanner/utils/Preference.java
package com.limetac.scanner.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class Preference {
    private String prefName = "limetac_prefs";

        private Context context = null;
        private SharedPreferences pref = null;

        private static String CURRENT_BLUETOOTH_NAME="CURRENT_BLUETOOTH_NAME";

        public Preference(final Context pContext) {
            this.context = pContext;
            pref = context.getSharedPreferences(prefName, Context.MODE_PRIVATE);
        }

        /**
         *
         * @param pRef
         * @param value
         */
        public void saveStringInPrefrence(final String pRef, final String value) {

            try {
                SharedPreferences.Editor editor = pref.edit();
                editor.putString(pRef, value);
                editor.apply();
            } catch (Exception e) {

            }
        }

        public void saveBooleanFlagInPrefrence(final String key, final boolean value) {

            try {
                SharedPreferences.Editor editor = pref.edit();
                editor.putBoolean(key, value);
                editor.apply();
            } catch (Exception e) {

            }
        }

        public boolean getBooleanFlagPrefrence(final String key, final boolean defaultValue) {
            return pref.getBoolean(key, defaultValue);
        }

        /**
         *
         * @param pRef
         * @param defaultValue
         * @return String
         */
        public String getStringPrefrence(final String pRef,
                                         final String defaultValue) {
            return pref.getString(pRef, defaultValue);
        }

        /**
         *
         * @param pRef
         * @param value
         */
        public void saveIntegerInPrefrence(final String pRef, final int value) {

            try {
                SharedPreferences.Editor editor = pref.edit();
                editor.putInt(pRef, value);
                editor.apply();
            } catch (Exception e) {

            }
        }

        /**
         *
         * @param pRef
         * @param defaultValue
         * @return String
         */
        public int getIntegerPrefrence(final String pRef, final int defaultValue) {
            return pref.getInt(pRef, defaultValue);
        }

        /**
         *
         * @param pRef
         * @param value
         */
        public void saveIntegerLongPrefrence(final String pRef, final long value) {

            try {
                SharedPreferences.Editor editor = pref.edit();
                editor.putLong(pRef, value);
                editor.apply();
            } catch (Exception e) {

            }
        }

        /**
         *
         * @param pRef
         * @param defaultValue
         * @return String
         */
        public long getLongPrefrence(final String pRef, final long defaultValue) {
            return pref.getLong(pRef, defaultValue);
        }



}

File: app/src/main/java/com/limetac/scanner/utils/Preference.kt
package com.limetac.scanner.utils

import android.content.Context
import android.content.SharedPreferences

class Preference(pContext: Context?) {
    private val prefName = "limetac_prefs"
    private var context: Context? = null
    private var pref: SharedPreferences? = null

    /**
     *
     * @param pRef
     * @param value
     */
    fun saveStringInPrefrence(pRef: String?, value: String?) {
        try {
            val editor = pref?.edit()
            editor?.putString(pRef, value)
            editor?.apply()
        } catch (e: Exception) {
        }
    }

    fun saveBooleanFlagInPrefrence(key: String?, value: Boolean) {
        try {
            val editor = pref?.edit()
            editor?.putBoolean(key, value)
            editor?.apply()
        } catch (e: Exception) {
        }
    }

    fun getBooleanFlagPrefrence(key: String?, defaultValue: Boolean): Boolean {
        return if(pref!=null) pref!!.getBoolean(key, defaultValue) else false
    }

    /**
     *
     * @param pRef
     * @param defaultValue
     * @return String
     */
    fun getStringPrefrence(
        pRef: String?,
        defaultValue: String?
    ): String? {
        return pref?.getString(pRef, defaultValue)
    }

    /**
     *
     * @param pRef
     * @param value
     */
    fun saveIntegerInPrefrence(pRef: String?, value: Int) {
        try {
            val editor = pref?.edit()
            editor?.putInt(pRef, value)
            editor?.apply()
        } catch (e: Exception) {
        }
    }

    /**
     *
     * @param pRef
     * @param defaultValue
     * @return String
     */
    fun getIntegerPrefrence(pRef: String?, defaultValue: Int): Int {
        return if (pref != null) pref!!.getInt(pRef, defaultValue) else -1
    }

    /**
     *
     * @param pRef
     * @param value
     */
    fun saveIntegerLongPrefrence(pRef: String?, value: Long) {
        try {
            val editor = pref?.edit()
            editor?.putLong(pRef, value)
            editor?.apply()
        } catch (e: Exception) {
        }
    }

    /**
     *
     * @param pRef
     * @param defaultValue
     * @return String
     */
    fun getLongPrefrence(pRef: String?, defaultValue: Long): Long {
        return if (pref != null) pref!!.getLong(pRef, defaultValue) else -1
    }

    companion object {
        private const val CURRENT_BLUETOOTH_NAME = "CURRENT_BLUETOOTH_NAME"
    }

    init {
        context = pContext
        pref = context?.getSharedPreferences(prefName, Context.MODE_PRIVATE)
    }
}